2024.12/apache/entrypoint.sh

```
@@ -39,8 +39,16 @@ file_env() {
     unset "$fileVar"
 }
 
+# initialize the email configuration
 sh /setup_msmtp.sh
 
+# ensure we have a logfile writeable by www-data
+FRIENDICA_LOGFILE=${FRIENDICA_LOGFILE:-/var/www/html/friendica.log}
+if sudo -u www-data test -w "$(dirname "$FRIENDICA_LOGFILE")"; then
+   touch "$FRIENDICA_LOGFILE"
+   chown www-data:www-data "$FRIENDICA_LOGFILE"
+fi
```

Two suggestions/options:
If it's just supposed to try to create it and not do anything except continue if it fails, one could skip the conditional and needing to install `sudo` and just try to create it:
```
touch "$FRIENDICA_LOGFILE" || true
chown --silent www-data:www-data "$FRIENDICA_LOGFILE" || true
```
(`--silent` should only be there if we want to suppress error messages)

Alternatively, instead of using `sudo` I'm guessing `su` or `runuser` are available out of the box. I'd suggest the latter if possible since it's deescalating privileges.

Edit: I guess the first approach doesn't really determine if the file is writable by `www-data`, so I guess the `runuser` / `su` approach is the better one of the two.
It would be something like this:
```sh
runuser --login www-data  --command "test -w \"$(dirname \"$FRIENDICA_LOGFILE\")"; then
```
...or
```sh
su --login www-data --command "test -w \"$(dirname \"$FRIENDICA_LOGFILE\")"; then
```

---

@mfxa, thank you for the comment.
I will try with the runuser. Not sure if su or runuser are available in the image. But I'll test.

---

@mfxa I tried. But none of the test worked correctly except for the solution with sudo.
The error in both other tests is: `This account is currently not available.`

---

Alright. Strange that the account is "currently not available".

Maybe `runuser` will work with `--user www-data` instead of `--login www-data`?

From the manpage:

--login: 
> Start the shell as a login shell with an environment similar to a real login:

--user:
> Run command with the effective user ID and group ID of the user name user.

Edit: But I think your Dockerfile idea is better.